### ISSUE TITLE
CFG construction does not have branches to catch blocks for unchecked exceptions

### ISSUE THREAD

**Issue by mernst** [typetools/checker-framework#1214]
The following code passes the Nullness Checker but fails with a null pointer exception at run time.

```java
public class TryCatch {

  public static void main(String[] args) {
    foo1().toString();
  }

  static String foo1() {
    String s = "not null";
    try {
      int data = 50 / 0;
    } catch (Exception e) {
      s = null;
    }
    return s;
  }
}
```

I suspect the problem is with dataflow's construction of the CFG -- the fact that `s` may be null is not reflected at the merge point after the catch clause.

The CFG construction doe.s not choose to add an outgoing edge to every operation, to reflect that the operation might throw an unchecked exception such as OutOfMemoryException.  Doing so would more accurately reflect the possible control flow, but might lead to many false positives.  (We should evaluate whether it really does lead to more false positives.)

A compromise solution would be that, if there is a catch block that catches unchecked exceptions, then edges are created to it from every operation that might throw any caught exception.  This still might lead to false positives, but it's bad style to write large try blocks or expansive catch clauses, so they should be rare.

**Comment by kanha95** [comment]
`import org.checkerframework.checker.nullness.qual.Nullable;

public class TryCatch {

  public static void main(String[] args) {
    foo1().toString();
  }

  static String foo1() {
    String s = "not null";
    try {
      int data = 50 / 0;
      
    } catch (Exception e) {
         try {
            s=null;
      } catch (Exception ee) {
      }
     
    }
    
    return s;
  }
}`

I also wish to add this test case professor. It also needs to consider the nested cases, i.e. another tryCatch block inside the catch block. 
We can also take into consideration the 'finally' block.

for example:-

`import org.checkerframework.checker.nullness.qual.Nullable;

public class TryCatch {

  public static void main(String[] args) {
    foo1().toString();
  }

  static String foo1() {
    String s = "not null";
    try {
      int data = 50 / 0;
      
    } catch (Exception e) {
         try {
           //some statements...
      } catch (Exception ee) {
      }
        finally {
        // missing warning
          s=null;
     }
    }
    
    return s;
  }
}`
